scripts/format.py

```diff
@@ -19,6 +19,15 @@
     print('you need to run `pip install black`', e)
     exit(-1)
 
+try:
+    ver = subprocess.check_output(('black', '--version'), text=True)
+    if '24.' not in ver:
```

Comment:
Shouldn't we check >=24 here?

Reply:
Yes, probably it's better >=24.
I even considered pinning to specific version (`24.1.0` is the latest) plus installing that specific one in CI.

If @Tishj has some preference, otherwise I will change to >=24.

Reply:
Actually, problem is that I am not sure how to check for >=24, given it's a string. I think `if '24.'` is fine given it still force everyone to update the formatter.

Reply:
`int(ver.split('.')[0]) >= 24`